easybuild/tools/filetools.py

```diff
@@ -1651,3 +1652,81 @@ def diff_files(path1, path2):
     file1_lines = ['%s\n' % l for l in read_file(path1).split('\n')]
     file2_lines = ['%s\n' % l for l in read_file(path2).split('\n')]
     return ''.join(difflib.unified_diff(file1_lines, file2_lines, fromfile=path1, tofile=path2))
+
+
+def load_hooks(hooks_path):
+    """Load defined hooks (if any)."""
+    hooks = []
+    if hooks_path:
+        (hooks_dir, hooks_filename) = os.path.split(hooks_path)
+        (hooks_mod_name, hooks_file_ext) = os.path.splitext(hooks_filename)
+        if hooks_file_ext == '.py':
+            _log.info("Importing hooks implementation from %s...", hooks_path)
+            (fh, pathname, descr) = imp.find_module(hooks_mod_name, [hooks_dir])
+            try:
+                # import module that defines hooks, and collect all functions of which name ends with '_hook'
+                imported_hooks = imp.load_module(hooks_mod_name, fh, pathname, descr)
+                for attr in dir(imported_hooks):
+                    if attr.endswith('_hook'):
+                        hook = getattr(imported_hooks, attr)
+                        if callable(hook):
+                            hooks.append(hook)
+                        else:
+                            _log.debug("Skipping non-callable attribute '%s' when loading hooks", attr)
+                _log.debug("Found hooks: %s", hooks)
+            except ImportError as err:
+                raise EasyBuildError("Failed to import hooks implementation from %s: %s", hooks_path, err)
+        else:
+            raise EasyBuildError("Provided path for hooks implementation should be location of a Python file (*.py)")
+    else:
+        _log.info("No location for hooks implementation provided, no hooks defined")
+
+    return hooks
+
+
+def find_hook(label, known_hooks, pre_step_hook=False, post_step_hook=False):
+    """
+    Find hook with specified label.
+
+    :param label: name of hook
+    :param known_hooks: list of known hooks
+    :param pre_step_hook: indicates whether hook to run is a pre-step hook
+    :param post_step_hook: indicates whether hook to run is a post-step hook
+    """
+    res = None
+
```

Comment:
The if below means that all hooks, except `start` and `end` need to have prefixes. Conceptually, I find it odd. If I create a `install_hook` I expect it to be executed after the install step. But it needs to be called `post_install_hook`. That way it is clear what it does, but I wonder if it makes sense to consider too `install_hook`.

Reply:
I feel `install_hook` is ambiguous. Some may assume it would be called as soon as the `install` step is triggered (equivalent with the current `pre_install_hook`), while your assumption is different.
To avoid guessing & confusion, I feel it's better to be explicit...

With your suggestion above in mind, you'd get an error like this when you define `install_hook`:

```
ERROR: 'install_hook' is not a valid hook; available hooks are: ..., pre_install_hook, post_install_hook, ...
```

Reply:
I think that's a reasonable approach.